eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailCommand.java

```diff
@@ -0,0 +1,287 @@
+package com.eternalcode.core.feature.jail;
+
+import com.eternalcode.annotations.scan.command.DescriptionDocs;
+import com.eternalcode.annotations.scan.feature.FeatureDocs;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.notice.NoticeService;
+import dev.rollczi.litecommands.annotations.argument.Arg;
+import dev.rollczi.litecommands.annotations.command.Command;
+import dev.rollczi.litecommands.annotations.context.Context;
+import dev.rollczi.litecommands.annotations.execute.Execute;
+import dev.rollczi.litecommands.annotations.permission.Permission;
+import java.time.Duration;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+@Command(name = "jail")
+@FeatureDocs(
+    name = "Jail",
+    permission = {"eternalcore.jail.bypass"},
+    description = "Permissions allow for you to setup jail location on your server, detain players, release players, and bypass jail."
+)
+public class JailCommand {
+
+    private final JailService jailService;
+    private final NoticeService noticeService;
+    private final PrisonerService prisonerService;
+
+    @Inject
+    JailCommand(JailService jailService, NoticeService noticeService, PrisonerService prisonerService) {
+        this.jailService = jailService;
+        this.noticeService = noticeService;
+        this.prisonerService = prisonerService;
+    }
+
+    @Execute(name = "setup")
+    @Permission("eternalcore.jail.setup")
+    @DescriptionDocs(description = "Define jail spawn area")
+    void executeJailSetup(@Context Player player) {
+        this.noticeService.create()
+            .notice(translation -> (this.jailService.isJailLocationSet()
+                ? translation.jailSection().jailLocationOverride()
+                : translation.jailSection().jailLocationSet()))
+            .player(player.getUniqueId())
+            .send();
+
+        Location location = player.getLocation();
+        this.jailService.setupJailArea(location);
+    }
+
+    @Execute(name = "setup")
+    @Permission("eternalcore.jail.setup")
+    @DescriptionDocs(description = "Define jail spawn area", arguments = "<location>")
+    void executeJailSetup(@Context Player player, @Arg Location location) {
+        this.noticeService.create()
+            .notice(translation -> (this.jailService.isJailLocationSet()
+                ? translation.jailSection().jailLocationOverride()
+                : translation.jailSection().jailLocationSet()))
+            .player(player.getUniqueId())
+            .send();
+
+        this.jailService.setupJailArea(location);
+    }
```

Comment:
```java
    @Execute(name = "setup")
    @Permission("eternalcore.jail.setup")
    @DescriptionDocs(description = "Define jail spawn area", arguments = "<location>")
    void executeJailSetup(@Context Player player) {
        this.jailService.setupJailArea(player.getLocation());

        this.noticeService.create()
            .notice(translation -> (this.jailService.isJailLocationSet()
                ? translation.jailSection().jailLocationOverride()
                : translation.jailSection().jailLocationSet()))
            .player(player.getUniqueId())
            .send();
    }
```

Reply:
This does not make sense

Reply:
?

Reply:
your method with `@Arg Location` will not work because world is not defined, it will throw an error

Reply:
I don't understand why u want to merge two commands into one. where are you collecting your `arguments = "<location>"` ??? What if player wants to use normal minecraft `-134 56 73` location?? 

Don't `?` me. Make it make sense

Reply:
> your method with `@Arg Location` will not work because world is not defined, it will throw an error

And I changed the order because if there is an error in setting the jail location, I will not send the message, which will be a sign to the user that an error occurred.  earlier, if an error occurred, a message about the jail setting would be sent and later an error might be sent anyway, it can be confusing

Reply:
> I don't understand why u want to merge two commands into one. where are you collecting your `arguments = "<location>"` ??? What if player wants to use normal minecraft `-134 56 73` location??
> 
> Don't `?` me. Make it make sense

https://github.com/EternalCodeTeam/EternalCore/pull/715#discussion_r1525926880